File: Basic Programming/Input-Output/Python/Minimize Cost.py
def minimize_cost(N, K, arr):
    res = 0
    i = 0

    while i < N:
        j = 1
        res += abs(arr[i])

        if arr[i] > 0:
            while j <= K and i + j < N:
                if arr[i + j] < 0:
                    res += arr[i + j]
                    j += 1
                else:
                    break

        i += j

    return abs(res)

# Example usage
N, K = map(int, input().split())
arr = list(map(int, input().split()))

result = minimize_cost(N, K, arr)
print(result)

File: Basic Programming/Input-Output/C++/Minimize Cost.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);

    int n, k;
    cin >> n >> k;

    vector<int> arr(n);
    for (int i = 0; i < n; ++i) {
        cin >> arr[i];
    }

    long long res = 0;
    int i = 0;

    while (i < n) {
        int j = 1;
        res += abs(arr[i]);

        if (arr[i] > 0) {
            while (j <= k && i + j < n) {
                if (arr[i + j] < 0) {
                    res += arr[i + j];
                    ++j;
                } else {
                    break;
                }
            }
        }

        i += j;
    }

    cout << abs(res) << endl;

    return 0;
}